src/libs/auth.js

```diff
@@ -153,6 +154,14 @@ authStore.subscribe(withErrorReporting('Error checking TOS', async (state, oldSt
   }
 }))
 
+authStore.subscribe(withErrorReporting('Error checking groups for timeout status', async (state, oldState) => {
+  if (oldState.registrationStatus !== state.registrationStatus) {
```

Comment:
In `initializeAuth`, there's a pattern for resetting dependent data during signout, which reduces the number of state transitions (and so reduces the possibility of edge cases). If we handled the reset there, then this code path could just check for the exact transition we care about, e.g. `oldState.registrationStatus !== 'registered' && state.registrationStatus === 'registered'`